From 1d769eeb7181e4c47bb3ed2e218b7e0956964f7b Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Fri, 19 Sep 2025 16:27:01 -0500
Subject: [PATCH 3/3] BLD: Disable non-standard CUDA arch checking

Local CUDA arch checking cannot handle -virtual,-real suffixes
which are valid values for CUDAARCHS.
---
 CMakeLists.txt | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d332c9bd..13193dca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -364,16 +364,6 @@ message(STATUS "CUDA .cu files compiler: ${USED_CUDA_COMPILER}")
 CUDA_find_supported_arch_values(CUDA_supported_archs ${USED_CUDA_COMPILER} ${CUDA_known_archs})
 message(STATUS "CUDA supported archs: ${CUDA_supported_archs}")
 
-set(CUDA_TARGET_ARCHS_SORTED ${CUDA_TARGET_ARCHS})
-list(SORT CUDA_TARGET_ARCHS_SORTED COMPARE NATURAL)
-CUDA_find_supported_arch_values(CUDA_targeted_archs ${USED_CUDA_COMPILER} ${CUDA_TARGET_ARCHS_SORTED})
-message(STATUS "CUDA targeted archs: ${CUDA_targeted_archs}")
-if (NOT CUDA_targeted_archs)
-  message(FATAL_ERROR "None of the provided CUDA architectures ({${CUDA_TARGET_ARCHS}})"
-      " is supported by ${USED_CUDA_COMPILER}, use one or more of: ${CUDA_supported_archs}")
-endif()
-
-CUDA_get_cmake_cuda_archs(CMAKE_CUDA_ARCHITECTURES ${CUDA_targeted_archs})
 message(STATUS "Generated CMAKE_CUDA_ARCHITECTURES: ${CMAKE_CUDA_ARCHITECTURES}")
 
 if(WIN32)
-- 
2.54.0

